### ISSUE TITLE
Kafka: "No subscriber found" when sending with Emitter

### ISSUE THREAD

**Issue by wicksim** [quarkusio/quarkus#10250]
**Describe the bug**
I'm getting `java.lang.IllegalStateException: No subscriber found for the channel XY` when trying to send messages with `Emitter`. In `application.properties` I've configured `mp.messaging.outgoing...` for this channel, so it should be sent to Kafka. [Documentation of 'SmallRye' states](https://smallrye.io/smallrye-reactive-messaging/smallrye-reactive-messaging/2/emitter/emitter.html): "You must have a @Incoming("prices") somewhere in your application (meaning a method consuming messages transiting on the channel prices), **or an outbound connector configured** to manage the prices channel (mp.messaging.outgoing.prices…​) ".

**Expected behavior**
`java.lang.IllegalStateException: No subscriber found for the channel XY` is thrown.

**Actual behavior**
Message should be sent to Kafka, even if there is no consumer at the moment.

**To Reproduce**
1. Checkout REST-Example-Projekt
2. Add SmallRye-Reactive-Messaging-Kafka-Dependency
3. Add outgoing-Configuration to `application.properties`
4. Add Emitter to REST-Endpoint
5. Start local Kafka
6. Start Quarkus

**Configuration**
```properties
mp.messaging.outgoing.xy.connector=smallrye-kafka
mp.messaging.outgoing.xy.value.serializer=io.quarkus.kafka.client.serialization.JsonbSerializer
```

**Screenshots**
None

**Environment (please complete the following information):**
 - Output of `uname -a` or `ver`: `Microsoft Windows [Version 10.0.17763.1217]`
 - Output of `java -version`: 
```
openjdk version "11.0.6" 2020-01-14 LTS
OpenJDK Runtime Environment Corretto-11.0.6.10.1 (build 11.0.6+10-LTS)
OpenJDK 64-Bit Server VM Corretto-11.0.6.10.1 (build 11.0.6+10-LTS, mixed mode)
```
 - GraalVM version (if different from Java): -
 - Quarkus version or git rev: `1.5.2.Final`
 - Build tool (ie. output of `mvnw --version` or `gradlew --version`): `Apache Maven 3.6.2 (40f52333136460af0dc0d7232c0dc0bcf0d9e117; 2019-08-27T17:06:16+02:00)`

**Additional context**
None

**Comment by wicksim** [comment]
Found the cause. I had 2 Producers, 1 `Emitter` and 1 `@Outgoing`. Both used the same channel-name, the name of the real Kafka-topic. So I had no explicit `.topic`-config in my `application.properties`. That doesn't seem to be a very good idea. I've now changed to 2 different channel-names and explicit `.topic`-configs pointing to the same Kafka-Topic. Now it works. So it was a misconfiguration from my side.

Only point left: Exception is not very self-explaining. Maybe it is worth to change that? But I've no idea, how complex this would be... Up to you, it's ok for me to close the issue.

**Comment by kenfinnigan** [comment]
How would the message be clarified, as it does indicate nothing is identified as subscribing to the emissions, either internally within the app or externally with Kafka?

**Comment by wicksim** [comment]
I agree, but in my case, a message "There are 2 publishers with same channel name" would have been helpful. So in this case, I first need to realize that I have 2 publishers with same channel name and first one some kind of "catches the subscriber" and there is no subscriber left for the second one. 

But if you think this is not necessary, just close it. My problem is solved and I hope to remember next time I'm trying to do that... 😄

**Comment by kenfinnigan** [comment]
We can close for now and re-evaluate if the situation continues to arise

**Comment by ManishJain-TomTom** [comment]
I faced a similar issue when I tried to configure an @Outgoing and Emitter which the same channel. To fix it, I created two channels pointing to same topic and used that.

However, now I am facing a different issue. Of these 2 publishers, a lot of times one is able to connect to the topic and other one reports that the topic is down. I am not able to understand why. 

Let's say I have publisher A with @Outgoing with topic XYZ. And then I have publisher B using emitter.send, and topic is XYZ again. If A is able to work with XYZ topic, then B should never say that XYZ is down. But that's the message I keep getting.

Reporting health down status: {"status":"DOWN","checks":[{"name":"SmallRye Reactive Messaging - readiness check","status":"DOWN","data":{"xxx-inbound":"[OK]","xxx":"[KO]","xxx-outbound":"[OK]"}}]}

Here XXX and XXX-Outbound connect with the same topic. I am connecting with Azure Event hubs using kafka protocol. Will apreciate any inputs on how to fix this intermittent issue.
